include/deal.II/base/index_set.h

```diff
@@ -385,8 +385,22 @@ class IndexSet
   size_type
   pop_front();
 
+  /**
+   * Return a vector with all indices contained in this IndexSet. This
+   * vector may of course be quite large if the IndexSet stores many
+   * indices. (This may be true even if the IndexSet itself does not
+   * take up much memory: IndexSet stores indices in a compressed
+   * format in which contiguous ranges of indices are only stored using
+   * pairs of indices.)
+   */
+  std::vector<size_type>
+  get_index_vector() const;
+
   /**
    * Fill the given vector with all indices contained in this IndexSet.
+   *
+   * This function is equivalent to calling get_index_vector() and
+   * assigning the result to the @p indices argument.
    */
   void
   fill_index_vector(std::vector<size_type> &indices) const;
```

Comment:
We deprecated the other fill functions - should this be deprecated too?

Reply:
I had no specific opinion about this, but am happy to do that if you'd like. Given that there are two thumbs up on your comment, I take that as a yes.

Reply:
Please do!